experimental/alp/python/alp/backend/codegen.py

```diff
@@ -0,0 +1,25 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from pathlib import Path
+from .utils import print_command, run_command
+
+
+def codegen(source, dest, scheduler="ilpmax"):
+    cmd = ["$IREE_LLVM_SANDBOX_BUILD_DIR/bin/llc"]
+    cmd.append(source)
+    cmd.append("-filetype=asm")
+    cmd.append("-O3")
+
+    # Scheduling options
+    if scheduler == "ilpmax":
+        cmd.append("--misched=ilpmax")
```

Comment:
interesting! I'd be interested in some details here, is this for your specific HW or something generic?

Reply:
This is generic. We basically found out that using different scheduler algorithms gives different performance (and many times the simpler shuffle algorithm works better). There are also some specific settings that apply to our particular case, but they were not generic enough to publish (e.g., disabling some optimizations in LLVM made things work better, somehow). My theory is that an HPC compiler like the one we are doing needs a tailored scheduler (that can spend a bit more time to schedule very large blocks), but many disagree with me. This file is the place where I do these experiments

Reply:
Would be great to factor this in a unified way that we could also experiment with in JIT mode.
Ideally we'd also want to experiment with techniques such as superoptimization but we are far from it atm.

Reply:
You guys always teach me new terms :) I was studying a lot of this, but I didn't know the technical term was "superoptimization"! Cool! Anyway, any suggestion to how to integrate in the JIT?

Reply:
I don't atm, it may be that a "JIT"-like flow needs to be built by passing files around and explicitly calling llvm, llc etc.
This is not a blocker for this revision to land, just something to think about.